src/bundle/Resources/public/js/scripts/admin.format.date.js

```diff
@@ -97,10 +97,29 @@
         );
     };
 
-    moment.fn.formatICU = function(format) {
-        const form = format.replace(formatICUEx, (icuStr) => {
+    const replaceFormat = function(format) {
+        return format.replace(formatICUEx, (icuStr) => {
             return typeof formatICUMap[icuStr] === 'function' ? formatICUMap[icuStr].call(this) : formatICUMap[icuStr];
         });
+    }
+
+    moment.fn.formatICU = function(format) {
+        /*
+            Splitting format by ' character in order to cut strings between ' ' as these strings should be escaped
+            according to ICU documentation: https://unicode-org.github.io/icu-docs/apidoc/released/icu4c/classSimpleDateFormat.html#details
+            At the end changing them to [ ] brackets for Moment.js
+        */
+        const form = format
+            .split('\'')
+            .map((partialStr, key) => {
```

Comment:
Maybe you can shorthand for map
`.map((partialStr, key) => key % 2 ? partialStr : replaceFormat(partialStr))`